### ISSUE TITLE
cp.js cpdirSyncRecursive doesn't correctly handle symlinks on windows

### ISSUE THREAD

**Issue by gatkinsNZ** [arturadib/shelljs#198]
The change https://github.com/arturadib/shelljs/commit/74ed9ca300a069092d5d3e62f8716a568aa25872 has broken recursive folder content copying on windows 8 when using symlinks. Instead of copying the symlink, it creates a junction link which links to the original file as if it was a directory not a file. This causes errors when using the files.

Came across this issue when using cordova CLI to do an Android build. See screenshot below - original folder is on the left, resulting folder on the right. There is some discussion of the issue of junction links here https://github.com/arturadib/shelljs/pull/104

![image](https://cloud.githubusercontent.com/assets/10623283/6910348/1e53f86a-d7ae-11e4-9458-fa8d5019ca3a.png)

**Comment by nfischer** [comment]
@gatkinsNZ do you know if this is unique to Windows 8, or does this apply to Windows 7 as well? If so, I can try this out on my Win7 machine.

**Comment by gatkinsNZ** [comment]
@nfischer This is an issue on windows 7 also - just tested and got the same result when using a .png file.

Just to clarify:
- I have a symlink to a file
- the file that is being copied is the symlink, but instead of the symlink being copied as is to the new location, a junction link is created instead, which links to the very original file as if it was a directory not a file

To get around this for now I have undone change 74ed9ca in my local file
e.g. 
/* fs.symlinkSync(symlinkFull, destFile, os.platform() === "win32" ? "junction" : null); */
fs.symlinkSync(symlinkFull, destFile);

**Comment by BYK** [comment]
This may also be resolved by #322.

**Comment by nfischer** [comment]
@gatkinsNZ would you be able to check out #322 (or check out the master branch once that is merged)? Reading your description, it looks like #322 will probably resolve it.

**Comment by nfischer** [comment]
@gatkinsNZ This is merged in master. If the issue is now fixed, feel free to close.

**Comment by gatkinsNZ** [comment]
@nfischer The issue I am striking is in cp.js, while #322 is a fix in ln.js. However the same solution I assume would work for cp() too..

**Comment by nfischer** [comment]
Ah, I see. Sounds like we'll have to open a new PR and add some more tests. Should be fairly straightforward, if someone wants to take it up.

**Comment by nfischer** [comment]
@gatkinsNZ is this now fixed on the latest version of shelljs? We did a lot of work surrounding symlinks.

**Comment by Goodwine** [comment]
I don't think this is entirely fixed. At least not for `shell.cp`. For context, I have a git repository which contains a symlink.

- Repo: https://github.com/sass/sass/tree/embedded-protocol-2.1.0
- Symlink: `spec/README.md` -> `spec/spec.md`

When I do `shell.cp(pathToRepo, newPath)` I get the error described above. \
Using `dir` (or `ln -l` in bash.exe) shows that the symlink `spec/README.md` now points to `spec/spec.md/` (absolute path tho).
